### PR TITLE
hypertrack-sdk-interface

### PR DESCRIPTION
**Pull request by Akr-Pandey**
1. Implementated 'com.google.firebase:firebase-messaging:23.0.7'
2. Remove new line from end of the file
3. Blocker implemention
4. Update 6.2.0 to 6.2.2

### CODE REVIEW

File: android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdk.java [hunk 1]
```diff
@@ -72,11 +73,326 @@ public HyperTrack initialize(String publishableKey) {
...
+        }
+        try {
+            String deviceId = sdkInstance.getDeviceID();
+            return deviceId;
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    public void hyperTrackRequestPermissions() {
```
Comment (pavel-kuznetsov-hypertrack) [inline android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdk.java:155]: what is that "hypertrack" prefix for?

File: android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java [hunk 2]
```diff
@@ -58,4 +65,245 @@ public void enableDebugLogging(PluginCall call) {
...
+            JSObject ret = new JSObject();
+            ret.put("value", "Availability listener added");
+            notifyListeners("availabilityStateChange", ret);
+        } catch (Exception e) {
+            call.reject(e.toString(),e);
+        }
+    }
+
+    @PluginMethod
+    public  void  removeAvailabilityListener(PluginCall call) {
```
Comment (pavel-kuznetsov-hypertrack) [inline android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java:256]: formatting issue

File: android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java [hunk 3]
```diff
@@ -58,4 +65,245 @@ public void enableDebugLogging(PluginCall call) {
...
+    @PluginMethod
+    public void getDeviceId(PluginCall call) {
+        implementation.print("getDeviceId called");
+        try {
+            String deviceId = implementation.getDeviceId();
+            JSObject ret = new JSObject();
+            ret.put("deviceId", implementation.print(deviceId));
+            call.resolve(ret);
+        } catch (Exception e) {
+            call.reject(e.toString(),e);
```
Comment (pavel-kuznetsov-hypertrack) [inline android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java:100]: formatting issue (please use autoformat)

File: android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java [hunk 4]
```diff
@@ -58,4 +65,245 @@ public void enableDebugLogging(PluginCall call) {
...
+        }
+    }
+
+    @PluginMethod
+    public void getDeviceId(PluginCall call) {
+        implementation.print("getDeviceId called");
+        try {
+            String deviceId = implementation.getDeviceId();
+            JSObject ret = new JSObject();
```
Comment (pavel-kuznetsov-hypertrack) [inline android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java:96]: please use meaningful variable names (e.g. result)

File: src/definitions.ts [hunk 5]
```diff
@@ -1,6 +1,195 @@
...
+    getAvailability():Promise<{ status:string }>;
+    /** Sets device's availability for nearby search. */
+    setAvailability(options:{ isAvailable:boolean }):Promise<void>;
+    /** Reflects tracking intent. */
+    isTracking():Promise<{ status: boolean }>;
+    /** Allows tracking hypertrack sdk  listener. */
+    addTrackingListener():Promise<void>;
+    /** Stops tracking hypertrack sdk  listener. */
+    removeTrackingListener():Promise<void>;
+    /** Allows search availability hypertrack sdk listener. */
```
Comment (pavel-kuznetsov-hypertrack) [inline src/definitions.ts:115]: WDYM by allows search?

File: src/definitions.ts [hunk 6]
```diff
@@ -1,6 +1,195 @@
...
+    addAvailabilityListener():Promise<void>;
```
Comment (pavel-kuznetsov-hypertrack) [inline src/definitions.ts:116]: why this method doesn't have parameter?

File: android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java [hunk 7]
```diff
@@ -58,4 +65,245 @@ public void enableDebugLogging(PluginCall call) {
...
+    }
+
+    @PluginMethod
+    public  void  removeAvailabilityListener(PluginCall call) {
+        implementation.print("removeAvailabilityListener called");
+        try {
+            implementation.removeAvailabilityListener(this);
+            JSObject ret = new JSObject();
+            ret.put("value", "Availability listener removed");
+            notifyListeners("availabilityStateChange", ret);
```
Comment (pavel-kuznetsov-hypertrack) [inline android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java:262]: why the listener is notified here?
What is "value" field for?

**Review by pavel-kuznetsov-hypertrack — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java (hunk 4 above)
Comment (Akr-Pandey) [reply android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java:96]: Sure

File: android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java (hunk 3 above)
Comment (Akr-Pandey) [reply android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdkPlugin.java:100]: I am already using that. I think, I have to do change something config.......let me check that

File: src/definitions.ts [hunk 8]
```diff
@@ -1,6 +1,195 @@
...
+   * @param options key-value pais of title & message properties.
+   */
+  setTrackingNotificationProperties(options: { title: string, message: string }): Promise<void>;
+  /**
+   * Adds listener for tracking state change
+   * 
+   * @param eventName 
+   * @param listenerFunc 
+   */
+  addListener(eventName: 'trackingStateChange', listenerFunc: StateChangeListener): Promise<PluginListenerHandle> & PluginListenerHandle;
```
Comment (pavel-kuznetsov-hypertrack) [inline src/definitions.ts:154]: @tulushev

File: src/definitions.ts [hunk 9]
```diff
@@ -1,6 +1,195 @@
...
+  /** Resolves device ID that could be used to identify the device. */
+  getDeviceId(): Promise<{ deviceId: string }>;
+  /** Resolves device's availability for nearby search. */
+  getAvailability(): Promise<{ status: string }>;
+  /** Sets device's availability for nearby search. */
+  setAvailability(options: { isAvailable: boolean }): Promise<void>;
+  /** Reflects tracking intent. */
+  isTracking(): Promise<{ status: boolean }>;
+  /** Allows tracking hypertrack sdk  listener. */
+  addTrackingListener(): Promise<void>;
```
Comment (pavel-kuznetsov-hypertrack) [inline src/definitions.ts:112]: @tulushev

File: android/build.gradle [hunk 10]
```diff
@@ -57,6 +57,7 @@ dependencies {
     implementation project(':capacitor-android')
     implementation "androidx.appcompat:appcompat:$androidxAppCompatVersion"
     implementation 'com.hypertrack:hypertrack:6.2.0'
+    implementation 'com.google.firebase:firebase-messaging:22.0.0'
```
Comment (pavel-kuznetsov-hypertrack) [inline android/build.gradle:60]: why not latest version (23.0.7)?

File: android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdk.java [hunk 11]
```diff
@@ -72,11 +73,323 @@ public HyperTrack initialize(String publishableKey) {
...
+        print("expected location argument " + coordinate);
+        JSONObject locationJsonObject = new JSONObject(coordinate);
+        Double latitude = locationJsonObject.getDouble("latitude");
+        Double longitude = locationJsonObject.getDouble("longitude");
+        Location expectedLocation = new Location(LocationManager.GPS_PROVIDER);
+        expectedLocation.setLatitude(latitude);
+        expectedLocation.setLongitude(longitude);
+        return expectedLocation;
+    }
 }
```
Comment (pavel-kuznetsov-hypertrack) [inline android/src/main/java/com/hypertrack/sdk/capacitor/HyperTrackSdk.java:395]: newline

**Review by pavel-kuznetsov-hypertrack — CHANGES_REQUESTED** [review]

**Pull request merged**